go-controller/pkg/node/base_node_network_controller_dpu.go

```diff
@@ -63,8 +63,8 @@ func (bnnc *BaseNodeNetworkController) delDPUPodForNAD(pod *corev1.Pod, dpuCD *u
 	podDesc := fmt.Sprintf("pod %s/%s for NAD %s", pod.Namespace, pod.Name, nadName)
 	klog.Infof("Deleting %s from DPU", podDesc)
 
-	if podDeleted {
-		// no need to unset connection status annotation if pod is deleted anyway
+	// no need to unset connection status annotation if pod is deleted anyway
+	if !podDeleted {
```

Comment:
this sounds like a path that can use a delete pod unit test for dpu workflow.. don't we have unit tests that ensure we are calling this path?